pkg/targets/adapter/solacetarget/adapter.go

```diff
@@ -0,0 +1,180 @@
+)
+
+// NewTarget adapter implementation
+func NewTarget(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClient cloudevents.Client) pkgadapter.Adapter {
+	logger := logging.FromContext(ctx)
+
+	mt := &pkgadapter.MetricTag{
+		ResourceGroup: targets.SolaceTargetResource.String(),
+		Namespace:     envAcc.GetNamespace(),
+		Name:          envAcc.GetName(),
+	}
+
+	metrics.MustRegisterEventProcessingStatsView()
+
+	env := envAcc.(*envAccessor)
+
+	idleTimeout := time.Duration(env.IdleTimeout) * time.Minute
+	connOption := amqp.ConnSASLAnonymous()
+
+	if env.SASLEnable {
+		connOption = amqp.ConnSASLPlain(env.Username, env.Password)
+	}
+
+	if env.TLSEnable {
+		tlsCfg := &tls.Config{}
+		if env.CA != "" {
+			addCAConfig(tlsCfg, env.CA)
+		}
+
+		if env.ClientCert != "" || env.ClientKey != "" {
+			if err := addTLSCerts(tlsCfg, env.ClientCert, env.ClientKey); err != nil {
+				logger.Panicw("Could not parse the TLS Certificates", zap.Error(err))
+			}
+		}
+
+		tlsCfg.InsecureSkipVerify = env.SkipVerify
+		connOption = amqp.ConnTLSConfig(tlsCfg)
+	}
+
+	amqpOpts := []amqp.ConnOption{
+		amqp.ConnIdleTimeout(idleTimeout),
+		connOption,
+	}
+
+	// Create amqp Client
+	amqpClient, err := amqp.Dial(env.URL, amqpOpts...)
+	if err != nil {
+		logger.Panicw("Could not create the amqp client", zap.Error(err))
+	}
+
+	return &solaceAdapter{
+		amqpClient: amqpClient,
+		amqpOpts:   amqpOpts,
+		queueName:  env.QueueName,
+
+		discardCEContext: env.DiscardCEContext,
+
+		ceClient: ceClient,
+		logger:   logger,
+
+		sr: metrics.MustNewEventProcessingStatsReporter(mt),
+	}
+}
+
+var _ pkgadapter.Adapter = (*solaceAdapter)(nil)
+
+type solaceAdapter struct {
+	amqpClient *amqp.Client
+	amqpOpts   []amqp.ConnOption
+	queueName  string
+	// createTopicIfMissing bool
+
+	discardCEContext bool
+
+	ceClient cloudevents.Client
+	logger   *zap.SugaredLogger
+
+	sr *metrics.EventProcessingStatsReporter
+}
+
+func (a *solaceAdapter) Start(ctx context.Context) error {
+	a.logger.Info("Starting Solace adapter")
+	return a.ceClient.StartReceiver(ctx, a.dispatch)
+}
+
+func (a *solaceAdapter) dispatch(event cloudevents.Event) cloudevents.Result {
+	ceTypeTag := metrics.TagEventType(event.Type())
+	ceSrcTag := metrics.TagEventSource(event.Source())
+
+	var msgVal []byte
+
+	start := time.Now()
+	defer func() {
+		a.sr.ReportProcessingLatency(time.Since(start), ceTypeTag, ceSrcTag)
+	}()
+
+	if a.discardCEContext {
+		msgVal = event.Data()
+	} else {
+		jsonEvent, err := json.Marshal(event)
+		if err != nil {
+			a.logger.Errorw("Error marshalling CloudEvent", zap.Error(err))
+			a.sr.ReportProcessingError(true, ceTypeTag, ceSrcTag)
+			return err
+		}
+		msgVal = jsonEvent
+	}
+
+	// Open a session
+	session, err := a.amqpClient.NewSession()
+	if err != nil {
+		log.Fatal("Creating amqp session")
+	}
+
+	// Create a sender
+	sender, err := session.NewSender(amqp.LinkTargetAddress(a.queueName))
+	if err != nil {
+		log.Fatal("Creating sender link:", err)
+	}
```

Comment:
Opening a session per dispatch dos not sound efficient. Should this be moved to Start?
Also, there are some log.Fatal ... Should that be using a.logger?
(Try not to panic on methods that are able to return an error)